### PR TITLE
Prebid 7: remove support for legacy FPD

### PR DESCRIPTION
**Pull request by dgirardi**
## Type of change

- [x] Other

## Description of change

This removes support for legacy FPD:

 - `setConfig('fpd')` is no longer converted to an equivalent `setConfig('ortb2')`
 - `adUnit.fpd` is no longer converted to an equivalent `adUnit.ortb2Imp`
 - `getLegacyFpd` and `getLegacyImpFpd`	have been moved away from `config` to standalone utils functions, and their use avoided when possible

I have kept the ortb2-to-legacy conversion logic available because of one, or possibly two adapters:

- `inmar` (@zandree-owneriq), sends legacy-formatted FPD verbatim to its backend; 
- `luponmedia` (@adxpremium) may have intended to do the same - but the implementation should not work even in version 6 (see code comments in this PR)

## Other information

Closes https://github.com/prebid/Prebid.js/issues/8283
Documentation PR: TBD, together with other Prebid 7 FPD-related changes (https://github.com/prebid/Prebid.js/issues/7651)

### CODE REVIEW

File: modules/inmarBidAdapter.js [hunk 1]
```diff
@@ -40,7 +41,7 @@ export const spec = {
       uspConsent: bidderRequest.uspConsent,
       currencyCode: config.getConfig('currency.adServerCurrency'),
       coppa: config.getConfig('coppa'),
-      firstPartyData: config.getLegacyFpd(bidderRequest.ortb2),
+      firstPartyData: getLegacyFpd(bidderRequest.ortb2),
```
Comment (patmmccann) [inline modules/inmarBidAdapter.js:44]: @zandree-owneriq we welcome your feedback on if you still need the data formatted this way or if you would prefer the newer ortb2 formatting?

File: modules/luponmediaBidAdapter.js [hunk 2]
```diff
@@ -453,6 +457,8 @@ function newOrtbBidRequest(bidRequest, bidderRequest, currentImps) {
     deepSetValue(data, 'ext.prebid.bidderconfig.0', bidderData);
   }
 
+  // TODO: bidRequest.fpd is not the right place for pbadslot - who's filling that in, if anyone?
+  // is this meant to be bidRequest.ortb2Imp.ext.data.pbadslot?
```
Comment (patmmccann) [inline modules/luponmediaBidAdapter.js:461]: @adxpremium fyi

File: src/utils/legacyFPD.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import {deepAccess, mergeDeep} from '../utils.js';
+
+/**
+ * Returns backwards compatible FPD data for modules
```
Comment (patmmccann) [inline src/utils/legacyFPD.js:4]: my only thought here is if it turns out only one adapter is using, eg inmar, would it be better to put this in the inmar adapter itself?

**Review by patmmccann — APPROVED** [review]

File: src/utils/legacyFPD.js (hunk 3 above)
Comment (dgirardi) [reply src/utils/legacyFPD.js:4]: yes, that would avoid extra weight for anyone who's not using inmar. I'll update if it turns out that luponmedia does not need the legacy conversion logic. Ideally, inmar would also drop the old format eventually.

**Comment by patmmccann** [comment]
#7796 for linkage

**Pull request merged**